### PR TITLE
Add simpler interface to build Crs

### PR DESCRIPTION
**Pull request by ibaned**
This addresses #991.

Note that PR #993 must be merged in first! (This PR builds on that one).

### CODE REVIEW

File: core/src/Kokkos_Crs.hpp [hunk 1]
```diff
@@ -328,6 +335,65 @@ void transpose_crs(
     FillCrsTransposeEntries<crs_type, crs_type> entries_functor(in, out);
 }
 
+template< class CrsType,
```
Comment (hcedwar) [inline core/src/Kokkos_Crs.hpp]: Unit test for this?

**Comment by ibaned** [comment]
Added a unit test and did a spot check with the following `test_all_sandia` on `kokkos-dev`:
```
cuda-8.0.44-Cuda_OpenMP-release
```

**Comment by hcedwar** [comment]
@ibaned : pull request has conflicts that must be resolved

**Comment by ibaned** [comment]
Pulled in the develop branch and resolved the conflict. Re-running spot check just in case.

**Comment by ibaned** [comment]
spot check passed again
```
cuda-8.0.44-Cuda_OpenMP-release build_time=367 run_time=550
```

**Pull request merged**